static/js/publisher/builds/components/repoConnect.js

```diff
@@ -0,0 +1,360 @@
+    this.setState(
+      {
+        selectedRepo: selectedRepo,
+        status: null
+      },
+      () => this.checkRepo(selectedRepo)
+    );
+  }
+
+  /**
+   * Handle organization select event
+   *
+   * @param organization
+   */
+  handleOrganiationSelect(selectedOrganization) {
+    this.setState({ selectedOrganization: selectedOrganization }, () =>
+      this.fetchRepoList()
+    );
+  }
+
+  /**
+   * Fetch repo list of the selected organization
+   *
+   */
+  fetchRepoList() {
+    const { selectedOrganization, user } = this.state;
+    let url = "";
+
+    if (selectedOrganization === user.login) {
+      url = "/publisher/github/get-repos";
+    } else {
+      url = `/publisher/github/get-repos?org=${selectedOrganization}`;
+    }
+    this.setState({
+      isRepoListDisabled: true,
+      status: LOADING
+    });
+
+    fetch(url)
+      .then(res => res.json())
+      .then(result => {
+        let newRepoList = result
+          .map(el => {
+            return { value: el.name };
+          })
+          .sort(this.sortByValue);
+        this.setState({
+          repoList: newRepoList,
+          isRepoListDisabled: false,
+          status: null
+        });
+      })
+      .catch(() => {
+        this.setState({
+          isRepoListDisabled: false,
+          status: ERROR
+        });
+      });
+  }
+
+  /**
+   * Check if repository is suitable for build
+   *
+   * @param selectedRepo
+   */
+  checkRepo(selectedRepo) {
+    const { snapName, repoList, selectedOrganization } = this.state;
+    if (selectedRepo && repoList.some(el => el.value === selectedRepo)) {
+      const url = `/${snapName}/builds/validate-repo?repo=${selectedOrganization}/${selectedRepo}`;
+
+      this.setState({
+        isRepoListDisabled: true,
+        status: LOADING
+      });
+
+      fetch(url)
+        .then(res => res.json())
+        .then(result => {
+          if (result.success) {
+            this.setState({
+              isRepoListDisabled: false,
+              status: SUCCESS
+            });
+          } else {
+            this.setState({
+              isRepoListDisabled: false,
+              status: result.error.type,
+              message: result.error.message,
+              yamlFilePath: result.error.yaml_location
+                ? result.error.yaml_location
+                : null
+            });
+          }
+        })
+        .catch(() => {
+          this.setState({
+            isRepoListDisabled: false,
+            status: ERROR
+          });
+        });
+    }
+  }
+
+  // Handle Refresh button click
+  handleRefreshButtonClick(e) {
+    e.preventDefault();
+    this.checkRepo(this.state.selectedRepo);
+  }
+
+  renderMessage() {
+    const {
+      message,
+      status,
+      selectedRepo,
+      selectedOrganization,
+      yamlFilePath,
+      snapName
+    } = this.state;
+    if (status === SNAP_NAME_DOES_NOT_MATCH) {
+      const splitDescription = message.split(". ");
```

Comment:
We shouldn't rely on the format of the message returned by the API. At some point someone might think "Oh this should say 'Yo - this don't match'" and change it in the backend. At which point this will no longer match and break this code path. It's better to just type what we want the message to say here.

Reply:
This still seems relevant - seems like you already had to update it based on the error in the API being different?

Reply:
In a perfect world you should see the same message for both js-enabled or js-disabled situations. However, the js-enabled message has a part that is bold, has a link, etc. and therefore is slightly different. Also in case there is a name mismatch, the message contains the name of whatever it is in the yaml file. Do we want to keep the messages separate (js-enabled and js-disabled)?